src/js/modules/sections/selectors.js

```diff
@@ -63,22 +33,38 @@ export const getTopLevelSections = createSelector(
 export const getTopLevelSectionsWithChildren = createSelector(
   [ getSections, getTopLevelSections ],
   (sections, topLevelSections) => {
-    let topLevelSectionsWithChildren = {};
-    Object.keys(topLevelSections).map(sectionSlug => {
-      const topLevelSection = topLevelSections[ sectionSlug ];
-      topLevelSectionsWithChildren[ sectionSlug ] = {
+    return Object.values(topLevelSections).reduce((acc, topLevelSection) => {
+      acc[ topLevelSection.id ] = {
         ...topLevelSection,
         subsections: Object.filter(sections, section => {
-          return section.parentSlug === section.slug;
+          if (section === topLevelSection) {
+            return false;
+          }
+          while (section.parentId !== null) {
+            section = sections[ section.parentId ];
```

Comment:
I...dunno. I'm having uneasy feelings about assigning an argument, but....there's no reason I think why not